### PR TITLE
2.1.0 fixes for nanoHUB tickets

### PR DESCRIPTION
**Pull request by stevvvn**
build.js unwebpacked is https://bitbucket.org/snyder13/citation-curation/src/4d99f366c16957dcd00651c50340acd81791c4dd/vue/?at=master

I wasn't sure if it was worth including or not

### CODE REVIEW

File: core/components/com_citations/curate/index.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php $mediaRoot = str_replace(JPATH_BASE, '', __DIR__); ?>
```
Comment (zooley) [inline core/components/com_citations/curate/index.php]: `JPATH_BASE` is deprecated.

`PATH_ROOT` gives the path to the `WEBROOT`
`PATH_CORE` gives the path to the `WEBROOT/core` directory
`PATH_APP` gives the path to the WEBROOT/app directory
`Component::path('com_citations');` gives the path to the component directory (no need for you to know if it's in app or core)

File: core/components/com_citations/site/controllers/curate.php [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+	 * See downloadTicket() for reasons why we want to do this.
+	 */
+	public function download($rest) {
+		if (isset($_GET['hash'])) {
+			$rest = [ $_GET['hash'] ];
+		}
+		if (!$rest) {
+			return \JError::raiseError(404, 'Not found');
+		}
+		$dbh = \JFactory::getDBO();
```
Comment (zooley) [inline core/components/com_citations/site/controllers/curate.php]: `JFactory` is deprecated. Use `App::get('db');`.

File: core/components/com_citations/site/controllers/curate.php [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+	/**
+	 * Get the data corresponding to a previous submission.
+	 * See downloadTicket() for reasons why we want to do this.
+	 */
+	public function download($rest) {
+		if (isset($_GET['hash'])) {
+			$rest = [ $_GET['hash'] ];
+		}
+		if (!$rest) {
+			return \JError::raiseError(404, 'Not found');
```
Comment (zooley) [inline core/components/com_citations/site/controllers/curate.php]: `\JError::raiseError` is deprecated. Use `App::abort` instead. Method arguments are the same (error number, message).

File: core/migrations/Migration20170412163541ComCitations.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ **/
+class Migration20170412163541ComCitations extends Base
+{
+        /**
+         * Up
+         **/
+        public function up()
+        {
+                foreach ([
+                'CREATE TABLE `jos_citations_visualizations` (
```
Comment (zooley) [inline core/migrations/Migration20170412163541ComCitations.php]: Don't hardcode database table prefixes. Should be `#__citations_visualizations`

File: core/components/com_citations/site/controllers/curate.php [hunk 5]
```diff
@@ -0,0 +1,251 @@
+<?php namespace Components\Citations\Site\Controllers;
```
Comment (zooley) [inline core/components/com_citations/site/controllers/curate.php]: It looks like this controller mostly outputs JSON? If so, probably should move to the API.

Comment (stevvvn) [inline core/components/com_citations/site/controllers/curate.php]: Mostly, but not entirely. I need at least the index route to output HTML

Comment (stevvvn) [inline core/components/com_citations/site/controllers/curate.php]: Also the functionality is very special-purpose for the SPA for the ERC work. I'm not sure the things it does are worth syndicating through the API

**Comment by stevvvn** [comment]
Crap, messed up the controller with that commit. Will push another shortly.

File: core/components/com_citations/site/controllers/curate.php (hunk 5 above)
Comment (zooley) [inline core/components/com_citations/site/controllers/curate.php]: To be blunt, that builds a case for it not being in the HUBzero core at all.

**Pull request closed without merging**